### PR TITLE
Feature/ik planner find iksolutions once

### PR DESCRIPTION
**Pull request by sjavdani**
-Removed SnapPlanner's PlanToEndEffectorPose, which didn't do any ik ranking
-Cleaned up IKPlanner to be in line with other planners
-Also added a new IK ranker that is useful if you know the next configuration you will plan to

Tested with a bunch of planning problems on jaco

### CODE REVIEW

**Comment by mkoval** [comment]
LGTM. Thanks @sjavdani!

**Comment by sjavdani** [comment]
Ping! Just making sure this wasn't forgotten about. Can we merge this into master and close?

File: src/prpy/planning/snap.py [hunk 1]
```diff
@@ -76,58 +76,6 @@ def PlanToConfiguration(self, robot, goal, **kw_args):
         """
         return self._Snap(robot, goal, **kw_args)
 
-    @ClonedPlanningMethod
-    def PlanToEndEffectorPose(self, robot, goal_pose, **kw_args):
```
Comment (mkoval) [inline src/prpy/planning/snap.py]: I am in favor of eventually removing `PlanToEndEffectorPose` from `SnapPlanner`, but I am concerned about breaking backwards compatibility in the process. 

The simplest solution is to implement a `PlanToEndEffectorPose` wrapper function on `SnapPlanner` that simply calls `_PlanToIK` on `IKPlanner`. In the process, you may want to rename `_PlanToIK` and (possibly) make it a `@classmethod`.

This is how I maintained backwards compatibility for CBiRRT's `PlanToEndEffectorOffset` method after introducing the `PlanToEndEffectorOffsetTSRAdapter` class in #311.

**Comment by mkoval** [comment]
The change overall looks good. On second thought, from my experience in #311, I have one new concern about backwards compatibility. @sjavdani can you address my comment above?

**Comment by psigen** [comment]
Why would this affect backwards compatibility if the previous
implementation did nothing?

**Comment by mkoval** [comment]
The previous implementation did *not* do nothing. However, it *was* missing some key features that @sjavdani needed on ADA (e.g. support for continuous joints, IK ranking). Thanks to @sjavdani, those features are now implemented in `IKPlanner`.

I am in favor of _eventually_ removing `PlanToEndEffectorPose` from `SnapPlanner` and encouraging people to switch to `IKPlanner(delegate_planner=SnapPlanner())`. But doing so will break backwards compatibility unless we take the steps I described above.

**Comment by psigen** [comment]
Is this a hypothetical break, or is there actually some code this affects?

I''d rather not add code bloat forever to support a deprecated use case no
one actually has or in lieu of a one line update in a few of our own
libraries.

Can we forward the method in SnapPlanner to an exception that details how
to update to Shervin's new method?

**Comment by mkoval** [comment]
It looks like @psigen is correct - I can't find a single instance of us using `PlanToEndEffectorPose` on `SnapPlanner` on HERB. We almost exclusively use `TSRLibrary` at this point.

@sjavdani is already switching to this code on ADA, so I retract my comment. We're better off removing the function, i.e. keeping the pull request as it is.

**Comment by psigen** [comment]
Thanks for taking it under consideration @mkoval.

**Comment by mkoval** [comment]
I will merge this once Travis is happy. An unrelated test failed, which I suspect was due to a transient load issue on the VM.

**Comment by mkoval** [comment]
Thanks @sjavdani. Sorry this fell through the cracks for so long.

**Comment by sjavdani** [comment]
No worries. As always, thanks for the careful and thoughtful consideration
that keeps us improving instead of bouncing and breaking things =)

**Pull request merged**